src/redisjson.rs

```diff
@@ -516,12 +538,28 @@ pub mod type_methods {
         // TODO remove in future versions
         if (encver == 2 && when == raw::Aux::Before as i32) {
             let map_size = raw::load_unsigned(rdb);
+            if is_io_error(rdb) {
+                return Status::Err as i32;
```

Comment:
`return Status::Err as i32;`

Should be added as a method on Status to_i32().
We already have the other way around:
```impl From<c_int> for Status {
    fn from(v: c_int) -> Self {
        Status::from_i32(v).unwrap()
    }
}
```